lib/internal/util.js

```diff
@@ -175,11 +175,20 @@ function slowCases(enc) {
   }
 }
 
-function emitExperimentalWarning(feature) {
+function emitExperimentalWarning(feature, code) {
   if (experimentalWarnings.has(feature)) return;
-  const msg = `${feature} is an experimental feature. This feature could ` +
-       'change at any time';
   experimentalWarnings.add(feature);
+  const { getOptionValue } = require('internal/options');
+  const opts = getOptionValue('--suppress-warnings');
+  const list = opts.split(',');
+  let found = false;
+  list.forEach((e) => {
+    if (e === code) found = true;
+  });
+  if (found) return;
+
+  const msg = `[${code}] ${feature} is an experimental feature. ` +
```

Comment:
This is not the right way to include the code. The emitWarning api already has an option for including the code correctly. Also, I would add this check into the emitWarning processing so it can be applied generically to all warning types.

Reply:
Not sure why this comment was marked resolved as the issue is still there.

Reply:
without knowing,  it was marked as resolved

Reply:
@jasnell - I have a doubt , can you confirm one thing please , the experimental warning `vm.measureMemory` API calls emitExperimentalWarning like this `emitExperimentalWarning('vm.measureMemory');`  https://github.com/nodejs/node/blob/22293eab481548c7dba3ffd320487d4f267b977b/lib/vm.js#L390 so the function takes a string and it matches with the `function emitExperimentalWarning(feature)`  like this https://github.com/nodejs/node/blob/51b43675067fafaad0abd7d4f62a6a5097db5044/lib/internal/util.js#L188 so if I'm not wrong this call does not include an option for codes. If the above file modification is not the right way to include the code, then how to do that? can you suggest me.